packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,588 @@
+  /// the [onValueChanged] callback and rebuild the segmented control with
+  /// the returned [Key] as the new [value] to update the currently selected
+  /// option.
+  ///
+  /// The [value] is the currently selected value for the segmented control.
+  /// If no [value] is provided, or the [value] is null, no widget will
+  /// appear as initially selected. This [value] must be the [Key] of one
+  /// of the widgets in the list of [children].
+  SegmentedControl({
+    Key key,
+    @required this.children,
+    @required this.onValueChanged,
+    this.value,
+  })  : assert(children != null),
+        assert(children.length >= 2),
+        assert(onValueChanged != null),
+        assert(value == null || children.keys.any((T child) => child == value)),
+        super(key: key);
+
+  /// The widgets in the segmented control.
+  ///
+  /// This attribute must not be null. The length of the list must be greater
+  /// than or equal to two.
+  final LinkedHashMap<T, Widget> children;
+
+  /// The [Key] value of the widget that is currently selected.
+  ///
+  /// This [Key] must be the [Key] of one of the widgets within [children].
+  /// If this attribute is null, no widget will be initially selected.
+  final T value;
+
+  /// The callback that is called when a new option is tapped.
+  ///
+  /// This attribute must not be null.
+  ///
+  /// The segmented control passes the newly selected widget's [Key] to the
+  /// callback but does not actually change state until the parent widget
+  /// rebuilds the segmented control with the new [value].
+  ///
+  /// The callback provided to [onValueChanged] should update the state of
+  /// the parent [StatefulWidget] using the [State.setState] method, so that
+  /// the parent gets rebuilt; for example:
+  ///
+  /// ## Sample code
+  ///
+  /// ```dart
+  /// class SegmentedControlExample extends StatefulWidget {
+  ///   @override
+  ///   State createState() => new SegmentedControlExampleState();
+  /// }
+  ///
+  /// class SegmentedControlExampleState extends State<SegmentedControlExample> {
+  ///   final List<Widget> children = <Widget>[
+  ///     const Text('Option 1', key: const ValueKey<int>(1)),
+  ///     const Text('Option 2', key: const ValueKey<int>(2)),
+  ///   ];
+  ///
+  ///   Key currentValue;
+  ///
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     return new Container(
+  ///       child:  new SegmentedControl(
+  ///         children: children,
+  ///         onValueChanged: (Key newValue) {
+  ///           setState(() {
+  ///             currentValue = newValue;
+  ///           });
+  ///         },
+  ///         value: currentValue,
+  ///       ),
+  ///     );
+  ///   }
+  /// }
+  /// ```
+  final ValueChanged<T> onValueChanged;
+
+  @override
+  _SegmentedControlState<T> createState() => _SegmentedControlState<T>();
+}
+
+class _SegmentedControlState<T> extends State<SegmentedControl<T>> {
+  ValueKey<int> _selectedValue;
```

Comment:
I think these should just be "T" types.

```dart
T _selectedValue;
T _pressedValue;
```

That way you can just use the keys from the `children`.